### PR TITLE
Move icon to dedicated component

### PR DESCRIPTION
**Pull request by pavelgruba**

### CODE REVIEW

File: js/renovation/icon.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+export const viewModelFunction = (model: Icon): IconViewModel => {
+    return {
+        cssClass: model.props.position !== 'left' ? 'dx-icon-right' : '',
+        props: model.props,
+        sourceType: getImageSourceType(model.props.source),
+    };
+};
+
+export const viewFunction = (viewModel: IconViewModel) => {
+    return (<Fragment>
```
Comment (LazyLahtak) [inline js/renovation/icon.tsx]: Should we decompose `viewModel`?
Something like
```tsx
export const viewFunction = ({ sourceType, cssClass, props: { source } }: IconViewModel) => {
```

Comment (pavelgruba) [reply js/renovation/icon.tsx]: I like it

File: testing/jest/button.tests.tsx [hunk 2]
```diff
@@ -259,49 +260,62 @@ describe('Button', () => {
...
-                    iconPosition: 'right',
+                    iconPosition: 'right'
                 });
 
                 const elements = button.find('.dx-button-content').children();
 
-                expect(button.hasClass('dx-button-icon-right')).toBe(true);
                 expect(elements.at(0).is('.dx-button-text')).toBe(true);
-                expect(elements.at(1).is('.dx-icon.dx-icon-test.dx-icon-right')).toBe(true);
+                expect(elements.at(1).is('Icon')).toBe(true);
```
Comment (MaximKudriavtsev) [inline testing/jest/button.tests.tsx]: Why is `Icon` string format here?

File: js/renovation/icon.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    viewModel: viewModelFunction,
+})
+export default class Icon extends JSXComponent<IconInput> {
+    // get sourceType() {
+    //     return getImageSourceType(this.props.source);
+    // }
+
+    // get cssClass() {
+    //     return this.props.position !== 'left' ? 'dx-icon-right' : '';
+    // }
```
Comment (MaximKudriavtsev) [inline js/renovation/icon.tsx:46]: Should we keep these comments?

File: js/renovation/icon.tsx (hunk 1 above)
Comment (MaximKudriavtsev) [reply js/renovation/icon.tsx]: I agreed with Andrey. We often use objects decomposition.

File: testing/jest/button.tests.tsx (hunk 2 above)
Comment (pavelgruba) [reply testing/jest/button.tests.tsx]: It's not a problem, but I'll fix it

File: js/renovation/icon.tsx (hunk 3 above)
Comment (pavelgruba) [reply js/renovation/icon.tsx:46]: I'd leave it for now. Once generator is ready to process new viewModel style I'll uncomment this code

**Review by roman-simionov — APPROVED** [review]

**Review by LazyLahtak — APPROVED** [review]

**Pull request merged**